lib/include/mxl/flow.h

```diff
@@ -136,8 +136,12 @@ extern "C"
         /// How many bytes in the grain are currently valid (commited).  This is typically used when writing slices.
         /// A grain is complete when commitedSize == grainSize
         uint32_t commitedSize;
+        /// absolute index of grain last written to this buffer
+        uint64_t grainIndex;
+        /// time stamp of grain last written to this buffer
+        uint64_t grainTimeStamp;
         /// User data space
-        uint8_t userData[4068];
+        uint8_t userData[4044];
```

Comment:
In order to align the true size of the struct without tail padding to 4K this needs to be 4048.

The actually size of the fields would leave even more space (4052 bytes) if it weren't for 4 bytes of padding before `grainIndex`. 

I think we are still in a phase where we don't need to maintain backwards compatibility so we could opt for recording these fields in a way that
1. makes sense and
2. avoids padding.

At a quick glance the padding could be avoided while still maintaining sensible grouping by sorting the fields like this:

* version
* size
* grainIndex
* grainTimeStamp
* flags
* payloadLocation
* deviceIndex
* grainSize
* commitedSize

To keep related fields aligned to 8 byte boundaries we could also opt to spend 64 bits on flags instead of 32.